pkgs/development/tools/build-managers/cmake/check-pc-files.sh

```diff
@@ -0,0 +1,18 @@
+cmakePcfileCheckPhase() {
+    while IFS= read -rd $'\0' file; do
+        grepout=$(grep --line-number '}//nix/store' "$file" || true)
+        if [ -n "$grepout" ]; then
+            {
+            echo "Broken paths found in a .pc file! $file"
+            echo "The following lines have issues (specifically '//' in paths)."
+            echo "$grepout"
+            echo "It is very likely that paths are being joined improperly."
+            echo 'ex: "${prefix}/@CMAKE_INSTALL_LIBDIR@" should be "@CMAKE_INSTALL_FULL_LIBDIR@"'
+            echo "Please see https://github.com/NixOS/nixpkgs/issues/144170 for more details."
+            exit 1
+            } 1>&2
+        fi
+    done < <(find "${!outputDev}" -iname "*.pc" -print0)
+}
+
+postFixupHooks+=(cmakePcfileCheckPhase)
```

Comment:
Can we add an option to opt-out of this?

Reply:
When there is a bug in your detection code it will be unconditionally break user builds. Some users may want to have "incorrect" files on purpose without having to workaround your code.

Reply:
the grep is already strict enough that there will be no non-broken false positives `}//nix/store`

Reply:
Well, I don't think I know enough about pkg-config to know if this is a robust pattern. If you are sure, go ahead.